api/libproject/devstart.rb

```diff
@@ -286,13 +289,15 @@ def drop_cloud_cdr(*args)
 end
 
 def connect_to_cloud_db(*args)
-  GcloudContext.new("connect-to-cloud-db", args, true).run do |ctx|
-    pw = ENV["WORKBENCH_DB_PASSWORD"]
-    # TODO Switch this to run_inline once Common supports redaction.
-    run_with_redirects(
-        "mysql -u \"workbench\" -p\"#{pw}\" --host 127.0.0.1 "\
-        "--port 3307 --database #{ENV["DB_NAME"]}",
-        to_redact=pw)
+  op = WbOptionsParser.new("connect-to-cloud-db", args)
+  CloudSqlProxyContext.new(GcloudContextV2.new(op)).run do |env_file_path|
```

Comment:
As a client, I'd slightly prefer having CloudSqlProxyContext extend from GcloudContextV2, or have a use_cloud_sql_proxy flag on GcloudContextV2... doing things this way gives me the opportunity to mess things up and not pass in GcloudContextV2 here.

Reply:
I see many uses for CloudSqlContext that have nothing to do with the proxy, so I don't want to conflate them. You don't even need a run block if you're not using the proxy. You can't mess it up though—try passing anything besides a context and you'll get an error.

Reply:
CloudSqlProxyContext uses gsutil, which in turn requires that you are logged into gcloud auth with your pmi-ops account... which is what GcloudContextV2 checks for us. I don't see in what situation we'd want the former without the latter.

Not a huge deal, but I'd make it non-optional.

Reply:
It is definitely not optional. Should we discuss? I can walk you through if needed.

Reply:
Right, so I was saying if it's not optional, don't make it an option -- have CloudSqlProxyContext automatically run the logic for GcloudContextV2, via one means or another.

Not a huge deal, but it's better if there are fewer ways for developers to screw things up.